Client/Frontend/Settings/AppSettingsOptions.swift

```diff
@@ -90,17 +105,45 @@ class SyncNowSetting: WithAccountSetting {
         return NSAttributedString(
             string: NSLocalizedString("Sync Now", comment: "Sync Firefox Account"),
             attributes: [
-                NSForegroundColorAttributeName: self.enabled ? UIColor.black : UIColor.gray,
+                NSForegroundColorAttributeName: self.enabled ? UIConstants.TableViewRowSyncTextColor : UIColor.gray,
                 NSFontAttributeName: DynamicFontHelper.defaultHelper.DefaultStandardFont
             ]
         )
     }
 
-    fileprivate let syncingTitle = NSAttributedString(string: Strings.SyncingMessageWithEllipsis, attributes: [NSForegroundColorAttributeName: UIColor.gray, NSFontAttributeName: UIFont.systemFont(ofSize: DynamicFontHelper.defaultHelper.DefaultStandardFontSize, weight: UIFontWeightRegular)])
+    fileprivate let syncingTitle = NSAttributedString(string: Strings.SyncingMessageWithEllipsis, attributes: [NSForegroundColorAttributeName: UIConstants.TableViewRowSyncTextColor, NSFontAttributeName: UIFont.systemFont(ofSize: DynamicFontHelper.defaultHelper.DefaultStandardFontSize, weight: UIFontWeightRegular)])
+    
+    func startRotateSyncIcon () {
+        imageView.layer.add(continuousRotateAnimation, forKey: nil)
+    }
+    
+    func stopRotateSyncIcon () {
+        imageView.layer.removeAllAnimations()
+    }
 
     override var accessoryType: UITableViewCellAccessoryType { return .none }
 
     override var style: UITableViewCellStyle { return .value1 }
+    
+    override var image: UIImage? {
+        if !AppConstants.MOZ_SHOW_FXA_AVATAR {
+            return nil
+        }
+        
+        let image = UIImage(named: "FxA-Sync")?.createScaled(CGSize(width: 20, height: 20))
+        guard let syncStatus = profile.syncManager.syncDisplayState else {
+            return image
+        }
+        
+        switch syncStatus {
+        case .inProgress:
+            self.startRotateSyncIcon()
+            return UIImage(named: "FxA-Sync-Blue")?.createScaled(CGSize(width: 20, height: 20))
```

Comment:
I wonder if we can do this with a transform, rather than switching image.

Reply:
@rnewman I tried updating this to use `UIView.transition` and switching depending on the animation state and the experience definitely not as smooth as the current one. I also played around with some other animation techniques but was getting diminishing returns on the time spent. That said, would the current animation be good enough to merge and get feedback then iterate?

Reply:
Yep, good enough!